cmd/create/execution.go

```diff
@@ -0,0 +1,123 @@
+)
+
+const (
+	executionShort = "Create execution resources"
+	executionLong  = `
+Create the executions for given launchplan in a project and domain.
+Currently the launchplan and where the execution needs to be launched , need to belong to same domain and project.
+
+There are three steps in generating an execution.
+
+- Generate the execution param yaml file containing in the default input parameters for the launch plan.
+- If the default parameters exist then you can modify the default param values in the generated yaml file.
+- Run the execution passing in the generated yaml file.
+
+The following command would generate core.basic.lp.go_greet.inputs.yaml file by using the launch plan provided in argument.
+This would use the latest version of the launch plan.
+::
+
+ bin/flytectl create execution core.basic.lp.go_greet  -d development  -p flytesnacks -g
+
+The generated file would look similar to this
+
+.. code-block:: yaml
+
+	inputs:
+	  am: false
+	  day_of_week: ""
+	  number: 0
+
+The generated file can be modified to change the parameter values and can be passed to command for execution
+::
+
+ bin/flytectl create execution core.basic.lp.go_greet -f core.basic.lp.go_greet.inputs.yaml  -d development  -p flytesnacks
+
+Usage
+`
+)
+
+//go:generate pflags ExecutionConfig --default-var executionConfig
+
+// ProjectConfig Config hold configuration for project create flags.
+type ExecutionConfig struct {
+	File     string `json:"file" pflag:",file for the execution params.If not specified defaults to <launchplan_name>.inputs.yaml"`
+	Version  string `json:"version" pflag:",version of the launch plan to be executed."`
+	GenParam bool   `json:"genParam" pflag:",bool flag to indicate the generation of params file."`
+}
+
+var (
+	executionConfig = &ExecutionConfig{}
+)
+
+func createExecutionCommand(ctx context.Context, args []string, cmdCtx cmdCore.CommandContext) error {
+	if len(args) == 0 {
+		return errors.New("pass in the launch plan to execute")
+	}
+	lpName := args[0]
+	var lp *admin.LaunchPlan
+	var err error
+	if lp, err = getLaunchPlan(ctx, lpName, cmdCtx); err != nil {
+		return err
+	}
+	var paramMap map[string]interface{}
+	if paramMap, err = getParamMap(lp); err != nil {
+		return err
+	}
+	var fileName string
+	// Generate the param file
+	if executionConfig.File != "" {
+		fileName = executionConfig.File
+	} else {
+		fileName = lpName + ".inputs.yaml"
+	}
+	if executionConfig.GenParam {
+		if err = writeParamsToFile(paramMap, fileName); err != nil {
+			return err
+		}
+		fmt.Printf("params written to file %v\n", fileName)
+		fmt.Printf("run followup command\nflytectl create execution %v -f %v\n", lpName, fileName)
+		return nil
+	}
+	var yamlData map[string]interface{}
+	if yamlData, err = readParamsFromFile(fileName); err != nil {
+		return err
+	}
+	// Convert to Literal map
+	var inputs *core.LiteralMap
+	if inputs, err = createLiteralMapFromParams(yamlData, lp.Spec.DefaultInputs.Parameters); err != nil {
```

Comment:
should this be `yamlData["inputs"]` As we just want to take the inputs and not the entire yaml

Reply:
The function

`readParamsFromFile`
only returns the params from the yaml file and hence we dont need to do that lookup here.
https://github.com/flyteorg/flytectl/pull/39/files#diff-0515dbf3e3fa779cd1f53566cbdada541292ad8641429d5bcb04ceba4117adcbR103

I will rename yamlData to paramsFromYaml to indicate this clearly.

Reply:
The function

`readParamsFromFile`
only returns the params from the yaml file and hence we dont need to do that lookup here.
https://github.com/flyteorg/flytectl/pull/39/files#diff-0515dbf3e3fa779cd1f53566cbdada541292ad8641429d5bcb04ceba4117adcbR103

I will rename yamlData to paramsFromYaml to indicate this clearly.

Reply:
can we change it to return a structure that you want to maintain in flytectl - rather than parsing as pure json.
like 
createExecution {
id:
inputs:
blah
}

Reply:
Done. Added the same structure we accept from command line for creating execution from file.